### DISCUSSION TITLE
Chess notation for actions

### DISCUSSION THREAD

**Discussion by AzerAfram** [deepmind/open_spiel#989] [Q&A] [answered]
Legal actions are represented as numbers. For example, a pawn move from a2 to a3 is represented as the number 89. This is a problem because I cannot tell what the move was from the number only. Any suggestions, explanations?

**Comment by lanctot** [comment, marked as answer]
C++ or python?

You can always use [State::ActionToString(int player, Action action)](https://github.com/deepmind/open_spiel/blob/e75bdf114de32c2211edf36443703a6e8846a3cb/open_spiel/spiel.h#L294) to get a string representation of the action. In Python, this would look like `state.action_to_string(player, action)`.

**Comment by AzerAfram** [comment]
Is there any way to get the coordinates of the piece that would be taking the action?

**Comment by lanctot** [comment]
Yep, you can use `chess::ActionToMove` here: https://github.com/deepmind/open_spiel/blob/e75bdf114de32c2211edf36443703a6e8846a3cb/open_spiel/games/chess.h#L120

and then the Move struct (whose definition is in chess_board.h) has the info you are looking for.

It'd help to answer the C++ vs. Python question if you have more follow-up questions; the answer for Python is that the function and data type is not exposed to Python (but can be easily added). I can offer to add this to show you how to add your own.

**Comment by AzerAfram** [comment]
Those attributes are what I am looking for. How can I use that function in Python?

**Reply by lanctot** [reply]
Currently, you can't: it needs to be exposed via the Chess-specific python functions, [here](https://github.com/deepmind/open_spiel/blob/master/open_spiel/python/pybind11/games_chess.cc), which is quite bare at the moment. 

Are you using OpenSpiel built from master? I'd be happy to expose it and sync to github in the next few days, with an example usage. But generally, see here: https://github.com/deepmind/open_spiel/blob/master/docs/developer_guide.md#adding-game-specific-functionality

**Reply by AzerAfram** [reply]
Thank you, but first I will try to get the from coordinates using the action as a string as well as the observation.

**Comment by AzerAfram** [comment]
I have changed my mind, that would be a very messy function. I am going to build from source, I would highly appreciate you exposing and syncing it. Thank you.

**Reply by lanctot** [reply]
Happy to do that, but I'm a bit loaded this week. Earliest it'd appear is Monday's sync to github, but I think I can do it by then.

**Reply by AzerAfram** [reply]
Take your time, and again, thank you.

**Reply by lanctot** [reply]
Ok, I've exposed some chess-specific functionality in https://github.com/deepmind/open_spiel/commit/453ab85f9dc94be8e2a3a01b12cf5821edfed141. This commit should give you an idea how to expose other things as you need (if expose more it'd be great to submit a PR to include it in master).

For an example of the specific thing you mentioned, see the only test in `games_chess_test.py`.

**Comment by AzerAfram** [comment]
Thank you!
